src/components/LogView/index.js

```diff
@@ -340,14 +371,19 @@ class LogView extends React.Component {
   render() {
     let j = 0;
     this.indexMap = {};
-
     this.filteredLines = this.props.lines.filter((line, i) => {
       if (!this.props.shouldPrintLine(this.props.bookmarks, line, this.props.filter, this.props.inverseFilter)) {
         return false;
       }
       this.indexMap[i] = j++;
       return true;
     });
+    this.highlightLines = this.filteredLines.filter((line) => {
+      if (!this.props.shouldHighlightLine(line, this.props.highlight, this.props.highlightLine)) {
```

Comment:
Ahh, now I see what you meant earlier about the highlight logic being in the filter logic. This works fine, but it's a performance risk, so let's do the following:
On line 373/374, put `this.state.highlightLines = [];` 
Then, on line 378, just before the return, we'll run this check, like so:
```
if (this.props.highlight.length > 0 && !this.props.shouldHighlightLine(line, this.props.highlight, this.props.highlightLine)) {
  this.state.highlightLines.push(line);
}
```
That should do it. Sorry about the misunderstanding.